hudi-common/src/main/java/org/apache/hudi/common/model/WriteOperationType.java

```diff
@@ -35,6 +35,7 @@
   // bulk insert
   BULK_INSERT("bulk_insert"),
   BULK_INSERT_PREPPED("bulk_insert_prepped"),
+  BULK_INSERT_DATASET("bulk_insert_dataset"),
```

Comment:
Need to understand why this is needed. so, we pick a different mode for the writer path I believe. We should use a config and not overload further if possible.